harness/determined/common/api/request.py

```diff
@@ -108,23 +88,34 @@ def do_request(
     timeout: Optional[Union[Tuple, float]] = None,
     max_retries: Optional[urllib3.util.retry.Retry] = None,
 ) -> requests.Response:
-    # If no explicit Authentication object was provided, use the cli's singleton Authentication.
-    if auth is None:
-        auth = authentication.cli_auth
-    if cert is None:
-        cert = certs.cli_cert
-
     if headers is None:
-        h = {}  # type: Dict[str, str]
+        h: Dict[str, str] = {}
     else:
         h = headers
 
+    if cert is None:
+        cert = certs.cli_cert
+
+    # set the token and username based on a perset precedence order.
```

Comment:
"perset"?  Should that be "preset"?

Maybe the comment should just say what the order is?